Userland/Applications/Presenter/Presentation.cpp

```diff
@@ -159,4 +159,19 @@ void Presentation::paint(Gfx::Painter& painter) const
     // FIXME: Fill the background with a color depending on the color scheme
     painter.clear_rect(painter.clip_rect(), Color::White);
     current_slide().paint(painter, m_current_frame_in_slide.value(), scale);
+
+    // FIXME: Move the footer format to user settings and make it configurable by the presentation file itself.
+    painter.draw_text(painter.clip_rect(), format_footer("{presentation_title}: {slide_title} ({slide_number}/{slides_total}), frame {slide_frame_number}"sv), Gfx::TextAlignment::BottomCenter);
+}
+
+String Presentation::format_footer(StringView format) const
+{
+    // FIXME: This is a totally crude find-replace hack.
+    // We don't need anything much more complicated, but we should allow for {{ and }} escapes somehow.
+    return format.replace("{presentation_title}"sv, title(), ReplaceMode::All)
+        .replace("{slide_title}"sv, current_slide().title(), ReplaceMode::All)
+        .replace("{author}"sv, author(), ReplaceMode::All)
+        .replace("{slides_total}"sv, String::number(m_slides.size()), ReplaceMode::All)
+        .replace("{slide_number}"sv, String::number(current_slide_number() + 1), ReplaceMode::All)
+        .replace("{slide_frame_number}"sv, String::number(current_frame_in_slide_number() + 1), ReplaceMode::All);
```

Comment:
We have a thing for that!
```suggestion
    StringBuilder builder;
    SourceGenerator generator{builder, '{', '}'};
    generator.set("presentation_title"sv, title());
    // ...
    generator.append(format);
    return generator.as_string();
```

Reply:
I feel stupid now